modules/phase_field/doc/content/modules/phase_field/Nucleation/DiscreteNucleation.md

```diff
@@ -27,16 +27,78 @@ adaptivity, and a postprocessor:
+[`DiscreteNucleationMap`](/DiscreteNucleationMap.md).
 
-## Example
+Multiple options exist for applyting the nucleus data to the variable fields in
+the simulation.
 
-An example input file is located at
+### Free energy penalty based nucleation
 
-!listing modules/phase_field/examples/nucleation/refine.i
+The free energy penalty based approaces eschew directly modifying conserved
+concentration and non-conserved order parameter fields. Instead they bias the
+thermodynamocs of the system in a way that drives the formation of nuclei at the
+locations provided by the discrete nucleation system. Simply stated the local
+free energy density is modified to make the nucleated state a lower energy
+state. This causes solute to diffuse toward the nucleation site (in the
+conserved case) brings about a change in a non-conserved order parameter field.
+Moose provides the [`DiscreteNucleation`](materials/DiscreteNucleation.md)
+material, which implements a simple harmonic form of such a penalty, but
+advanced users are free to implement their own free energy modifications
+following the example set in this material.
+
+#### Conserved order parameters
+
+Use a [`DerivativeSumMaterial`](/DerivativeSumMaterial.md) to add the nucleation
+free energy penalty to the physical free energy contributions of the system
+(which is utilized by a Cahn-Hilliard kernel).
+
+#### Non-Conserved order parameters
+
+Use either a [`DerivativeSumMaterial`](/DerivativeSumMaterial.md) to add the
+nucleation free energy penalty to the physical free energy contributions of the
+system, or add an additional [`AllenCahn`](/AllenCahn.md) kernel using only the
+nucleation penalty energy to the variable influenced by the nucleation process.
+The latter approach also works for hardcoded free energies that do not utilize
+the [Function Material](/phase_field/FunctionMaterials.md) approach.
+
+### Direct order parameter modification
+
+In some cases, such as polycrystalline models used in conjunction with grain
+tracker, direct modification of non-conserved order parameters can be a feasible
+approach.
+
+Nucleation of new grains in a polycrystalline model with
+[GrainTracker](/GrainTracker.md) requires the use of reserved (or staging) order
+parameters in which new grain nuclei can be inserted and successively get piched
+up and remapped onto the evolving order parameters.
+
+Apply both a [DiscreteNucleationForce](/DiscreteNucleationForce.md) and a
+[Reaction](/Reaction.md) kernel to a reserved order parameter and set the
+`hold_time` in the inserter to `0`. This will cause the modification of the
+reserved OP during only a single timestep, which will be enough for the
+GrainTracker to pick up and remap the new nucleus.
+
+## Advanced subjects
+
+!media phase_field/refine.gif style=width:50%;margin-left:20px;float:right;
+       caption=Targeted mesh refinement prior to nucleus insertion.
+
+The use of mesh and timestep adaptivity is supported through the
```

Comment:
I am cautious about this bit about mesh and timestep adaptivity.  Yes, they do work numerically, and that is excellent.  However, I wonder if there should be a note added about the difficulties/pitfalls that can occur with using various adaptivities.  Obviously this can become a significant discussion beyond the scope of the documentation.  Maybe it would be a good idea  to point to the paper I wrote where I discuss some simple basics (section 3.2 and on)? A. M. Jokisaari et al., Computational Materials Science 112 (2016) 128–138.

Reply:
> "[...] time or mesh adaptivity do not appear to affect the simulated kinetics."

:-/ I think this would be overkill here. What I could add though is a discussion of the upper bound limits for timestep and mesh coarseness. Beyond just discussing this I could actually add another limit to the timestep to remain in the rare event regime. I will do that in a separate PR.

Reply:
Okay, that sounds reasonable.